pkg/runtime/wfengine/activity.go

```diff
@@ -184,6 +194,8 @@ loop:
 			if !t.Stop() {
 				<-t.C
 			}
+			// Activity execution failed with non-recoverable error. Record metrics
+			diag.DefaultWorkflowMonitoring.ExecutionEvent(ctx, "dapr", diag.Activity, diag.StatusFailed, 0)
```

Comment:
It's a retriable error as mentioned in the comment. Also add the time elapsed in the event